src/main/java/com/commercetools/sync/producttypes/utils/AttributeDefinitionsUpdateActionUtils.java

```diff
@@ -252,16 +251,25 @@ private static boolean haveSameAttributeType(
      * Otherwise, if there are no new attribute definitions, then an empty list
      * is returned.
      *
-     * @param newAttributeDefinitionDrafts  the list of new {@link AttributeDefinitionDraft}s.
-     * @param oldAttributeDefinitionNameMap a map of names to AttributeDefinition of the old list
-     *                                      of attribute definition.
+     * @param oldAttributeDefinitions      the list of old {@link AttributeDefinition}s.
+     * @param newAttributeDefinitionDrafts the list of new {@link AttributeDefinitionDraft}s.
+     *
      * @return a list of attribute definition update actions if there are new attribute definition that should be added.
      *         Otherwise, if the attribute definitions are identical, an empty optional is returned.
      */
     @Nonnull
     private static List<UpdateAction<ProductType>> buildAddAttributeDefinitionUpdateActions(
-        @Nonnull final List<AttributeDefinitionDraft> newAttributeDefinitionDrafts,
-        @Nonnull final Map<String, AttributeDefinition> oldAttributeDefinitionNameMap) {
+        @Nonnull final List<AttributeDefinition> oldAttributeDefinitions,
+        @Nonnull final List<AttributeDefinitionDraft> newAttributeDefinitionDrafts ) {
+
+        final Map<String, AttributeDefinition> oldAttributeDefinitionNameMap =
```

Comment:
What is not clear in here?

````java
/**
     * Checks if there are any new attribute definition drafts which are not existing in the
     * {@code oldAttributeDefinitionNameMap}. If there are, then "add" attribute definition update actions are built.
     * Otherwise, if there are no new attribute definitions, then an empty list
     * is returned.
     *
     * @param oldAttributeDefinitions      the list of old {@link AttributeDefinition}s.
     * @param newAttributeDefinitionDrafts the list of new {@link AttributeDefinitionDraft}s.
     *
     * @return a list of attribute definition update actions if there are new attribute definition that should be added.
     *         Otherwise, if the attribute definitions are identical, an empty optional is returned.
     */
````

Reply:
`oldAttributeDefinitionNameMap` is not a passed param anymore. It's rather an inner variable of the method. So checking the Javadoc without looking inside the method will probably be very confusing :)

Reply:
Hmm, I changed the parameter, imo this doc is clear for this method. https://github.com/commercetools/commercetools-sync-java/pull/316/commits/edebdcc035d032c88b2abaeaf5df6619767c4c88